src/core/operations/ECDSASignatureConversion.mjs

```diff
@@ -0,0 +1,106 @@
+/**
+ * @author cplussharp
+ * @copyright Crown Copyright 2021
+ * @license Apache-2.0
+ */
+
+import Operation from "../Operation.mjs";
+import r from "jsrsasign";
+
+/**
+ * ECDSA Sign operation
+ */
+class ECDSASignatureConversion extends Operation {
+
+    /**
+     * ECDSASignatureConversion constructor
+     */
+    constructor() {
+        super();
+
+        this.name = "ECDSA Signature Conversion";
+        this.module = "Ciphers";
+        this.description = "Convert an ECDSA signature between hex, asn1 and json.";
+        this.infoURL = "https://wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm";
+        this.inputType = "string";
+        this.outputType = "string";
+        this.args = [
+            {
+                name: "Input Format",
+                type: "option",
+                value: [
+                    "Auto",
+                    "ASN.1 HEX",
+                    "Concat HEX",
+                    "JSON"
+                ]
+            },
+            {
+                name: "Output Format",
+                type: "option",
+                value: [
+                    "ASN.1 HEX",
+                    "Concat HEX",
+                    "JSON"
+                ]
+            }
+        ];
+    }
+
+    /**
+     * @param {string} input
+     * @param {Object[]} args
+     * @returns {string}
+     */
+    run(input, args) {
+        let inputFormat = args[0];
+        const outputFormat = args[1];
+
+        // detect input format
+        if (inputFormat === "Auto") {
+            if (input.substr(0, 2) === "30" && r.ASN1HEX.isASN1HEX(input)) {
+                inputFormat = "ASN.1 HEX";
+            } else if (input.indexOf("{") !== -1) {
+                inputFormat = "JSON";
+            } else {
+                inputFormat = "Concat HEX";
+            }
+        }
+
+        // convert input to ASN.1 hex
+        let signatureASN1Hex;
+        switch (inputFormat) {
+            case "ASN.1 HEX":
+                signatureASN1Hex = input;
+                break;
+            case "Concat HEX":
+                signatureASN1Hex = r.KJUR.crypto.ECDSA.concatSigToASN1Sig(input);
+                break;
+            case "JSON": {
```

Comment:
This doesn't seem like it represents the JWK format.  What format is this?  I would have expected something like [this](http://localhost:8080/#recipe=ECDSA_Signature_Conversion('JSON','ASN.1%20HEX')&input=eyJrdHkiOiJFQyIsImNydiI6IlAtMjU2IiwieCI6IkUzc1hsbjlyT0FaYXlHVjdKM2QxNERHQzJmVUhEdXdzY0JZa0JkaHQza2siLCJ5IjoiWk1GSC1GOXdFZ1EzYTBIQjJaZ1hSTVp1N1VzVEYwTnJ0dzI4VDN1Q2hCRSIsImQiOiJPY3JNckZ3dmhaeDJUNDNNbDFvbXhsSTFOaDJTcExtbGYyT3ZveXFrdDU0In0&ieol=CRLF) to work but it doesn't seem to.

I'd expect this format to prefer existing specifications, but either way there should be some sanity checks to ensure that both 'r' and 's' exist in the JSON.

Reply:
"r" and "s" are the two parts of an ECDSA signature. I'm not sure if the JSON format of the signature is used somewhere directly but some libraries want the signature as two parameters!

Reply:
My mistake, I believe this is caused by my confusion above between 'signature' and 'key' conversion.  This seems like a valid format for inputting these values.  Could we still add a check to make sure that the JSON is (a) valid and (b) Has both 'r' and 's' attributes.